plugins/postgres/dbt/include/postgres/macros/relations/materialized_view/rename.sql

```diff
@@ -1,3 +1,11 @@
 {% macro postgres__get_rename_materialized_view_sql(relation, new_name) %}
+  {% if relation is not none %}
+    {% set database = relation.database %}
+    {% set schema = relation.schema %}
+    {% set to_relation = adapter.get_relation(database=database, schema=schema, identifier=new_name) %}
+    {% if to_relation is not none %}
+        {{ adapter.rename_relation(from_relation=relation, to_relation=to_relation) }}
+    {% endif %}
+  {% endif %}
```

Comment:
Currently this seems the most correct attempt we've made

When in a `debug` session of a `dbt run --full-refresh` run we see
```
ipdb> locals().keys()
dict_keys(['l_1_relation', 'l_1_new_name', 't_2', 'l_1_database', 'l_1_schema', 'l_1_adapter', 'l_1_to_relation', 'l_1_debug', 'concat', 'context', 'environment', 'missing', 'resolve', 't_1', 'undefined'])
ipdb> l_1_relation
<PostgresRelation "dbt"."dbt_mcknight"."orders">
ipdb> l_1_new_name
'orders__dbt_backup'
ipdb> l_1_database
'dbt'
ipdb> l_1_schema
'dbt_mcknight'
ipdb> l_1_to_relation
ipdb> l_1_to_relation
```

but on run end we are getting error 
```Internal Error
  Cache inconsistency detected: in rename, new key _ReferenceKey(database='dbt', schema='dbt_mcknight', identifier='orders') already in cach```